pkg/util/condition_test.go

```diff
@@ -56,3 +58,102 @@ func TestIsConditionsEqual(t *testing.T) {
+	}{
+		{
+			name: "Case 1: add new condition to ouputConditions.",
+			inputCondition: metav1.Condition{
+				Type:               "Ready",
+				Status:             metav1.ConditionTrue,
+				Reason:             fakeEventSuccessReason,
+				Message:            fakeSuccessMessage,
+				LastTransitionTime: metav1.Unix(int64(1679494373), int64(0)),
+			},
+			ouputConditions: &[]metav1.Condition{},
+			expectedConditions: &[]metav1.Condition{
+				{
+					Type:               "Ready",
+					Status:             metav1.ConditionTrue,
+					Reason:             fakeEventSuccessReason,
+					Message:            fakeSuccessMessage,
+					LastTransitionTime: metav1.Unix(int64(1679494373), int64(0)),
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Case 2: If the condition status is the equal, the LastTransitionTime time is the equal.",
+			inputCondition: metav1.Condition{
+				Type:               "Ready",
+				Status:             metav1.ConditionTrue,
+				Reason:             fakeEventSuccessReason,
+				Message:            fakeSuccessMessage,
+				LastTransitionTime: metav1.Unix(int64(1679494373), int64(0)),
+			},
+			ouputConditions: &[]metav1.Condition{
+				{
+					Type:               "Ready",
+					Status:             metav1.ConditionTrue,
+					Reason:             fakeEventSuccessReason,
+					Message:            fakeSuccessMessage,
+					LastTransitionTime: metav1.Unix(int64(200000000000), int64(0)),
+				},
+			},
+			expectedConditions: &[]metav1.Condition{
+				{
+					Type:               "Ready",
+					Status:             metav1.ConditionTrue,
+					Reason:             fakeEventSuccessReason,
+					Message:            fakeSuccessMessage,
+					LastTransitionTime: metav1.Unix(int64(200000000000), int64(0)),
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Case 3: if the condition status is not equal, the LastTransitionTime time will be change.",
+			inputCondition: metav1.Condition{
+				Type:               "Ready",
+				Status:             metav1.ConditionFalse,
+				Reason:             fakeEventSuccessReason,
+				Message:            fakeSuccessMessage,
+				LastTransitionTime: metav1.Unix(int64(1679496434), int64(0)),
+			},
+			ouputConditions: &[]metav1.Condition{
+				{
+					Type:               "Ready",
+					Status:             metav1.ConditionTrue,
+					Reason:             fakeEventSuccessReason,
+					Message:            fakeSuccessMessage,
+					LastTransitionTime: metav1.Unix(int64(1679494373), int64(0)),
+				},
+			},
+			expectedConditions: &[]metav1.Condition{
+				{
+					Type:               "Ready",
+					Status:             metav1.ConditionFalse,
+					Reason:             fakeEventSuccessReason,
+					Message:            fakeSuccessMessage,
+					LastTransitionTime: metav1.Unix(int64(1679496434), int64(0)),
+				},
+			},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			meta.SetStatusCondition(tt.ouputConditions, tt.inputCondition)
```

Comment:
We don't need this unit test, as this function is maintained by Kubernetes and there is already a unit test at:
https://github.com/kubernetes/kubernetes/blob/d2be69ac11346d2a0fab8c3c168c4255db99c56f/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions_test.go#L27